src/sonic-config-engine/minigraph.py

```diff
@@ -517,6 +520,31 @@ def filter_acl_mirror_table_bindings(acls, neighbors, port_channels):
 
     return acls
 
+def select_mmu_profiles(profile, platform, hwsku):
+    """
+        Select MMU files based on the device metadata attribute - SonicQosProfile
+        if no QosProfile exists in the minigraph, then no action is needed.
+        if a profile exists in the minigraph,
+            - create a dir path to search 1 level down from the base path.
+            - if no such dir path exists, no action is needed.
+            - if a dir path exists, check for the presence of each file from
+              the copy list in the dir path and copy it over to the base path.
+    """
+    if not profile:
+        return
+
+    files_to_copy = ['pg_profile_lookup.ini', 'qos.json.j2', 'buffers_defaults_t0.j2', 'buffers_defaults_t1.j2']
+
+    path = os.path.join('/usr/share/sonic/device', platform, hwsku)
+
+    dir_path = os.path.join(path, profile.upper())
```

Comment:
Not sure we want to do profile.upper() here, this will limit our options to up case only.

Reply:
I added that based on the naming for the subfolders where everything is in upper case. Should we rely on the exact name in the minigraph?

Reply:
I think we should rely on exact match.